app/include/zmk/keymap.h

```diff
@@ -11,9 +11,11 @@ typedef uint32_t zmk_keymap_layers_state_t;
 uint8_t zmk_keymap_layer_default();
 zmk_keymap_layers_state_t zmk_keymap_layer_state();
 bool zmk_keymap_layer_active(uint8_t layer);
+bool zmk_keymap_layer_active_with_state(uint8_t layer, zmk_keymap_layers_state_t state_to_test);
```

Comment:
Sorry, one last thing I missed. This is only used internally in `keymap.c`, we shouldn't publish this method yet, unless it's actually needed.